### PR TITLE
solve issues marked by SonarCloud as high severity

### PR DESCRIPTION
**Pull request by hrajagers**

### CODE REVIEW

**Comment by hrajagers** [comment]
@MRVermeulenDeltares  or @ralph-peelen-deltares  Would you be able to review these changes. I've tried to address all "high severity" issues identified by SonarCloud. Unfortunately, it doesn't seem to state whether they have actually been merged into the trunk ...

File: dfastmi/batch/Face.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def filter_faces_by_face_condition(
     inode : numpy.ndarray
         Array of length K2 containing the indices of the nodes to keep [-].
     """
-    iface = numpy.where(condition)[0]
+    iface = numpy.nonzero(condition)[0]
```
Comment (ralph-peelen-deltares) [inline dfastmi/batch/Face.py:102]: Do you only want to get the 'True' values?

File: dfastmi/batch/Projection.py [hunk 2]
```diff
@@ -36,6 +36,194 @@
 import dfastmi.batch.Face
 
 
+def project_one_xy_point_beyond_segment(
```
Comment (ralph-peelen-deltares) [inline dfastmi/batch/Projection.py:39]: Unit test this method

File: dfastmi/batch/Projection.py [hunk 3]
```diff
@@ -36,6 +36,194 @@
...
+            xyp[1] - p0[1] - alpha * (p1[1] - p0[1])
+        ) ** 2
+        dist2end = dist - dist2link
+        if dist2end > 100:
+            dist = 1e20
+
+    return dist, sgn
+
+
+def project_one_xy_point_onto_segment(
```
Comment (ralph-peelen-deltares) [inline dfastmi/batch/Projection.py:87]: is this unit tested

File: dfastmi/batch/Projection.py [hunk 4]
```diff
@@ -36,6 +36,194 @@
...
+        # if it's actually closer than the node ...
+        if dist2link < dist:
+            # update the closest point information
+            dist = dist2link
+            s = s0 + alpha * (s1 - s0)
+
+    return dist, sgn, s
+
+
+def project_one_xy_point_onto_line(
```
Comment (ralph-peelen-deltares) [inline dfastmi/batch/Projection.py:152]: no unit test?

Comment (ralph-peelen-deltares) [reply dfastmi/batch/Projection.py:152]: is a private method, please update accordingly

File: dfastmi/batch/core.py [hunk 5]
```diff
@@ -692,6 +694,51 @@ def _log_length_estimate(report: TextIO, slength: float) -> None:
     )
 
 
+def write_report_discharges(
```
Comment (ralph-peelen-deltares) [inline dfastmi/batch/core.py:697]: is a private method, please update accordingly

File: dfastmi/io/DataTextFileOperations.py [hunk 6]
```diff
@@ -134,37 +134,25 @@ def read_xyc(
             else:
                 colnames = ["X", "Y"]
             if hasHeader:
```
Comment (ralph-peelen-deltares) [inline dfastmi/io/DataTextFileOperations.py:136]: hasHeader = ralph implementatie... please use snake_case has_header

File: dfastmi/io/DataTextFileOperations.py [hunk 7]
```diff
@@ -134,37 +134,25 @@ def read_xyc(
...
-                    )
-                else:
-                    P = pandas.read_csv(
-                        filename,
-                        names=colnames,
-                        skipinitialspace=True,
-                        header=0,
-                        delim_whitespace=True,
-                    )
+                header = 0
```
Comment (ralph-peelen-deltares) [inline dfastmi/io/DataTextFileOperations.py:137]: header = 0 if hasHeader else None

File: dfastmi/cli.py [hunk 8]
```diff
@@ -68,114 +68,118 @@ def interactive_mode(src: TextIO, rivers: RiversObject, reduced_output: bool) ->
...
+        all_done = run_interactive_mode_once(
+            src, rivers, reduced_output, report, have_files
+        )
+
+    ApplicationSettingsHelper.log_text("end")
+    ApplicationSettingsHelper.log_text("end", file=report)
+    report.close()
+
+
+def run_interactive_mode_once(
```
Comment (ralph-peelen-deltares) [inline dfastmi/cli.py:80]: is private?

File: dfastmi/cli.py [hunk 9]
```diff
@@ -437,26 +414,162 @@ def interactive_get_discharges(
...
+    if all_q and apply_q[2] and discharges[2] is not None:
+        new_discharges[2] = interactive_check_discharge(
+            src, 3, discharges[2], stages[i_prev], new_discharges[i_prev]
+        )
+        all_q = new_discharges[2] is not None
+
+    return all_q, new_discharges
+
+
+def write_report_data(
```
Comment (ralph-peelen-deltares) [inline dfastmi/cli.py:486]: private?

**Review by ralph-peelen-deltares — CHANGES_REQUESTED** [review]
Please process remarks

File: dfastmi/io/DataTextFileOperations.py (hunk 6 above)
Comment (hrajagers) [reply dfastmi/io/DataTextFileOperations.py]: Code updated.

File: dfastmi/io/DataTextFileOperations.py (hunk 7 above)
Comment (hrajagers) [reply dfastmi/io/DataTextFileOperations.py]: syntax adopted

**Review by hrajagers — COMMENTED** [review]
@ralph-peelen-deltares All comments processed. Code updated.

**Review by ralph-peelen-deltares — APPROVED** [review]

**Pull request merged**